visvis2/material/_base.py

```diff
@@ -1,6 +1,97 @@
+from python_shader import Struct, vec2, mat4
+import numpy as np
+
+from .._wrappers import BufferWrapper
+
+
+# Definition of struct with standard info,
+# provided to each world-object as uniform at slot 0.
+stdinfo_type = Struct(
+    world_transform=mat4,
+    cam_transform=mat4,
+    projection_transform=mat4,
+    physical_size=vec2,
+    logical_size=vec2,
+)
+
+
+def array_from_shader_type(spirv_type):
+    """ Get a numpy array object from a SpirV type from python-shader.
+    """
+    return np.asarray(spirv_type())
+
+
 class Material:
     def __init__(self):
         self.dirty = True
-        self.shaders = {}
-        self.uniforms = {}
-        self.primitive_topology = None
+
+    def get_wgpu_info(self, obj):
+        """ Return a list of high level pipeline descriptions.
+        These can be compute or render pipelines.
+
+        The default implementation returns a single render pipeline description,
+        using vertex_shader, fragment_shader, and primitive_topology from the material,
+        and index and vertex buffers from the geometry.
```

Comment:
It feels really wrong to be working with WorldObject and Geometry inside of a Material. Why not solve this in the renderer?